### ISSUE TITLE
Impossible to assign log parser configuration to a log file

### ISSUE THREAD

**Issue by hayos** [otros-systems/otroslogviewer#541]
It is impossible to assign a log parser configuration to a log file in Otros in OtrosLogViewer 1.4.12.

At least I am not able to find the way how to do it in 45 min, either in the program itself or in the documentation. I was only able to create a parser configuration, but not to select it.

The result is that I do not see a single row from my log files.

Since I am a developer with more than 30 years of experience, your user interface is apparently so weird at this point, that it makes the product unusable.

**Comment by otrebski** [comment]
Hi @hayos .
On welcome screen you have following options:
 - `Open log files` - open log files with automatic selection of log parser. If more than one log format is recognised, log parser will be chosen by first matching rule
- `Merge log files` - advance option. You can select log files from different directories, choose to tail from begging of end of file, select specific log parser for every log file.

Is it helpful?

**Comment by ragafoo** [comment]
in OtrosLogViewer 1.4.12.
I am having the same issue, I cannot launch the application with my json .pattern file, how do we get the app to launch with a specific logparser via the cmd line? 

In windows this launches the correct file but no parsing, is there an additional parameter I can provide?

"**C:\Users\bobby\Desktop\olv-1.4.12>** olv.sh D:\test\Logging\BUFFER-Routine-JSON.log"

**Comment by thothothotho** [comment]
> Hi @hayos .
> On welcome screen you have following options:
> 
>     * `Open log files` - open log files with automatic selection of log parser. If more than one log format is recognised, log parser will be chosen by first matching rule
> 
>     * `Merge log files` - advance option. You can select log files from different directories, choose to tail from begging of end of file, select specific log parser for every log file.
> 
> 
> Is it helpful?

I filled a similar issue, I realize that this thread exists.  It is  counter intuitive to use "Merge log files" when you don't want to merge anything.  Maybe you should rename it to "Advanced open".

The UI is broken anyway olv-1.4.12.  After adding a file in the "Merge log files", I still cannot change the parser.

Automatic detection is good when you have 100% success.  Otherwise, you should provide in the open
panel a way to choose the parser (and if the panel remembers the last chosen parser, that will be good enough for 99.9% of the users)

**Comment by p91paul** [comment]
The issue here is that even in the "Merge log files" window olv tries to use autodetection. This fails if in the same log file the first lines use a different log pattern than the rest of the file, or for files with very long lines where the autodetection simply never finishes. Anyway the combobox only shows patterns it *thinks are applicable* to that file, and doesn't allow to apply another one. There is no option to **force** using a specific pattern.

**Comment by maddingo** [comment]
I created a PR #566. This adds a menu entry for each log parser it finds at start-up. 
Another improvement could be to add a command line parameter that lets you set the log parser instead of the auto-detection.

**Comment by pplupo** [comment]
I agree with that. Besides, I have a log file here that triggers a message "Can't parse server.log" in the footer of the open log dialog. There is no option to select a parser. Not even using the "merge log" option.

**Comment by svennissel** [comment]
In Version 1.5 there is option to choose a parser manually
